lib/ansible/plugins/lookup/fileglob.py

```diff
@@ -32,8 +33,16 @@ def run(self, terms, variables=None, **kwargs):
         ret = []
         for term in terms:
             term_file = os.path.basename(term)
-            dwimmed_path = self.find_file_in_search_path(variables, 'files', os.path.dirname(term))
-            if dwimmed_path:
+            term_dir = os.path.dirname(term)
+            if term_dir.startswith('~') or term_dir.startswith(os.path.sep):
+                basepath = unfrackpath(to_text(term_dir))
+                candidates = [basepath]
```

Comment:
I guess I could do the same thing as above (building a one-item list as candidates for paths starting in ~ or /) in path_dwim_get_candidates, which would then leave path_dwim_relative_stack as merely a) getting the candidates) and b) the check for existence of all candidates in the result of a).